openfl/component/director/director.py

```diff
@@ -164,21 +175,19 @@ def remove_experiment_data(self, experiment_name: str, caller: str):
 
     def collaborator_health_check(self, *, collaborator_name: str,
                                   is_experiment_running: bool,
-                                  valid_duration: int) -> bool:
+                                  valid_duration: int) -> int:
         """Accept health check from envoy."""
-        is_accepted = False
         shard_info = self._shard_registry.get(collaborator_name)
         if not shard_info:
-            logger.error(f'Unknown shard {collaborator_name}')
-            return is_accepted
-        is_accepted = True
+            raise Exception(f'Unknown shard {collaborator_name}')
 
         shard_info['is_online']: True
         shard_info['is_experiment_running'] = is_experiment_running
         shard_info['valid_duration'] = valid_duration
         shard_info['last_updated'] = time.time()
+        hc_period = self.settings.get('envoy_health_check_period', ENVOY_HEALTH_CHECK_PERIOD)
```

Comment:
You have 2 variables here: `hc_period` and `valid_duration` and they always have the same relation y = 2*x.
So could we remove one?

Reply:
`valid_duration` - this variable is sent by the envoy and means how long the sent information can be considered valid
`health_check_period` - the director sends this variable to the envoy to configure. It's possible to change its value during the life of the service and, because of this, `valid_duration` depends on the preciously set value.

Reply:
It was refactored